src/ray/raylet/raylet.cc

```diff
@@ -96,8 +96,7 @@ Raylet::Raylet(instrumented_io_context &main_service,
 Raylet::~Raylet() {}
 
 void Raylet::Start() {
-  RAY_CHECK_OK(RegisterGcs());
-
+  TryToGetAgentInfoAndRegisterGcs();
```

Comment:
can you just move all implementation into `RegisterGcs ` instead of creating a new method? `TryToGetAgentInfo` is just an implementation detail

Reply:
Good idea!